src/Security/Authentication/Negotiate/src/NegotiateOptions.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Authentication.Negotiate
+{
+    /// <summary>
+    /// Options class provides information needed to control Negotiate Authentication handler behavior
+    /// </summary>
+    public class NegotiateOptions : AuthenticationSchemeOptions
+    {
+        /// <summary>
+        /// The object provided by the application to process events raised by the negotiate authentication handler.
+        /// The application may use the existing NegotiateEvents instance and assign delegates only to the events it
+        /// wants to process. The application may also replace it with its own derived instance.
+        /// </summary>
+        public new NegotiateEvents Events
+        {
+            get { return (NegotiateEvents)base.Events; }
+            set { base.Events = value; }
+        }
+
+        /// <summary>
+        /// Indicates if Kerberos credentials should be persisted and re-used for subsquent anonymous requests.
+        /// This option must not be used if connections may be shared by requests from different users.
+        /// The default is false.
+        /// </summary>
+        public bool PersistKerberosCredentials { get; set; } = false;
+
+        /// <summary>
+        /// Indicates if NTLM credentials should be persisted and re-used for subsquent anonymous requests.
+        /// This option must not be used if connections may be shared by requests from different users.
+        /// The default is true.
+        /// </summary>
+        public bool PersistNtlmCredentials { get; set; } = true;
+
+        // For testing
+        internal INegotiateStateFactory StateFactory { get; set; } = new ReflectedNegotiateStateFactory();
+    }
```

Comment:
This is internal and the only place setting this property is test code. So basically you have a setter only for the purpose of testing. Isn't this what DI is designed to avoid?

Reply:
You're right, but I'd have to make INegotiateStateFactory and INegotiateState public as well. We don't want to do that when this contract exists only for testing.